server/stycoon/reports.go

```diff
@@ -238,6 +260,8 @@ func (game *Game) fillAllReportsSinceSeasonStart() {
 
 func (game *Game) getDataReports() DataReports {
 	var tickReports DataReports
+	game.Ready.Lock()
```

Comment:
You should use Read locks when only reading from the map - example here https://github.com/gdg-garage/space-tycoon/blob/master/server/handlers/game.go#L25 
That's probbly making it very slow

Reply:
Btw, the whole GetData is locked on the line I sent - from there we call GetData and getDataReports. IMHO the whole logic is locked already